src/QMCHamiltonians/ObservableHelper.h

```diff
@@ -109,21 +87,19 @@ class ObservableHelper
   void addProperty(std::vector<float>& p, const std::string& pname, hdf_archive& file);
   void addProperty(std::vector<TinyVector<float, OHMMS_DIM>>& p, const std::string& pname, hdf_archive& file);
 
+  void write(const value_type* const first_v, hdf_archive& file);
 
-  void write(const value_type* first_v, const value_type* first_vv);
-
-  /**
-   * Check if hdf5 handlers are valid after a call to ObservableHelper::open
-   * true: opened, false: not opened
-   */
-  bool isOpened() const noexcept;
-
+  ///starting index
+  hsize_t lower_bound = 0;
 private:
-  ///true: hdf5 handlers are valid via open, false: otherwise
-  bool isopened = false;
-
-  ///closes remaining hdf5 handlers in destructor if isopened = true
-  void close();
+  /// "file pointer" for h5d_append
+  hsize_t current = 0;
+  ///name of this observable
+  std::vector<std::string> group_name;
```

Comment:
Could you say a bit more? "vector" seems surprising. Also the variable name is not plural.

Reply:
`hdf_archive` isn't in the same top-level group when `ObservableHelper::open(...)` and `ObservableHelper::write(...)` are called. We need to store a path containing multiple groups. I initially tried writing a longer string where each group was separated by `/` but HDF5 wouldn't create multiple groups at once. I found it easier to iterate through a vector of strings than than do the string manipulation to generate these substrings.

Essentially, I'm iterating over a a HDF5 path similar to how you might do on the fileystem with [std::filesystem::path]( 
https://en.cppreference.com/w/cpp/filesystem/path/begin)

Reply:
Yes. That is useful info. Could you expand the comment saying the groups that a patch consists of is stored here.